lib/reversetunnel/localsite.go

```diff
@@ -602,23 +671,55 @@ func (s *localSite) chanTransportConn(rconn *remoteConn, dreq *sshutils.DialReq)
 	if err != nil {
 		if markInvalid {
 			rconn.markInvalid(err)
+			// If not serving any connections close and remove this connection immediately.
+			// Otherwise, let the heartbeat handler detect this connection is down.
+			if rconn.activeSessions() == 0 {
+				s.removeRemoteConn(rconn)
+				return nil, trace.NewAggregate(trace.Wrap(err), rconn.Close())
+			}
```

Comment:
Is this removal actually necessary?  IIUC removal here means that we still might hit the case where a conn may continue to hb, but isn't routable.  I think we're going to have an easier time if removal is tightly coupled with closure.